File: kotlin/junit/cheatsheet/java/src/main/java/hirabay/junit5/cheetsheet/java/ParentComponent.java
package hirabay.junit5.cheetsheet.java;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@RequiredArgsConstructor
@Component
public class ParentComponent {
    private final ChildComponent childComponent;

    public String greeting(String name) {
        return childComponent.greeting(name);
    }
}

File: kotlin/junit/cheatsheet/kotlin/src/main/kotlin/hirabay/junit5/cheatsheet/ParentComponent.kt
package hirabay.junit5.cheatsheet

import org.springframework.stereotype.Component

@Component
class ParentComponent(private val childComponent: ChildComponent) {
    fun greeting(name: String): String {
        return childComponent.greeting(name)
    }

    suspend fun greetingSuspend(name: String): String {
        return childComponent.greetingSuspend(name)
    }
}